### DISCUSSION TITLE
[ZK Stack] - Failure at ecosystem init

### DISCUSSION THREAD

**Discussion by thomas-nguy** [zkSync-Community-Hub/zksync-developers#823] [ZK Stack]
### Team or Project

Cronos Labs

### Module Affected

- [X] System contracts (era-system-contracts)
- [X] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.82.0

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

Hi 

I am following the guide to setup the dev environment 
https://github.com/matter-labs/zksync-era/tree/main/zkstack_cli

When running the command `zkstack ecosystem init`, I am facing an error when building the contracts

```
    Status:
│      exit status: 1
│    Stdout:
│      2024-12-12T04:43:47.621114Z ERROR foundry_compilers_artifacts_solc::sources:
│      error="<ZK_HOME>/zksync-era/contracts/l1-contracts/lib/openzeppelin-contracts-v4/contracts/token/ERC20/IERC20.sol":
│      Not a directory (os error 20)
│
│
│    Stderr:
│      Error: failed to resolve file:
│      "<ZK_HOME>/zksync-era/contracts/l1-contracts/lib/openzeppelin-contracts-v4/contracts/token/ERC20/IERC20.sol":
│      Not a directory (os error 20); check configured remappings
│      --> <ZK_HOME>/zksync-era/contracts/l1-contracts/contracts/bridge/L1ERC20Bridge.sol
│      @openzeppelin/contracts-v4/token/ERC20/IERC20.sol
│
```

It seems that the foundry mapping is not working well on my machine.

Looking at the `l1-contracts/lib` folders, it seems the package use relative mapping to the real directory
https://github.com/matter-labs/era-contracts/tree/46d75088e7ddb534101874c3ec15b877da1cb417/l1-contracts/lib

For example:

https://github.com/matter-labs/era-contracts/blob/46d75088e7ddb534101874c3ec15b877da1cb417/l1-contracts/lib/openzeppelin-contracts-v4

I can bypass this error if I rewrite the `foundry.toml` with absolute path here
https://github.com/matter-labs/era-contracts/blob/46d75088e7ddb534101874c3ec15b877da1cb417/l1-contracts/foundry.toml

But it does not seem to be a sustainable solution

Does anyone faces the same issues or any clue how to fix?

Thanks

### Expected Behavior

Able to compile

### Current Behavior

Throw error at contract compilation

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It appears that the issue you're encountering is related to the path resolution for the OpenZeppelin contracts in your Foundry setup. This is a common issue when the relative paths in the `foundry.toml` file do not correctly resolve to the actual directories where the contracts are stored.

Here are a few steps you can try to resolve this issue:

1. **Verify the Remappings in `foundry.toml`:** Ensure that the remappings in your `foundry.toml` file are correctly pointing to the actual paths of the OpenZeppelin contracts. You might need to adjust these paths to match the structure of your local filesystem.

2. **Use Absolute Paths Temporarily:** As you mentioned, using absolute paths can resolve the issue temporarily. To make this a more sustainable solution, you can script the generation of the `foundry.toml` file to dynamically insert absolute paths based on your environment.

3. **Clear Foundry's Cache:** Sometimes, clearing the cache can help resolve path-related issues. You can do this by running:
   ```
   forge clean
   ```

4. **Update Foundry:** Ensure that you are using the latest version of Foundry, as updates often include bug fixes that might resolve your issue.

5. **Consult the ZKsync Developers Community:** If the issue persists, it might be helpful to discuss it with other developers who might have faced similar issues. You can post your question in the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) for more in-depth assistance.

If these steps do not resolve the issue, it might be beneficial to look into any specific configurations or environmental factors that could be affecting the path resolution in your setup.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by sarahschwartz** [comment]
Hi @thomas-nguy, do you have the latest version of [`zksync-foundry`](https://foundry-book.zksync.io/getting-started/installation) installed?

**Reply by thomas-nguy** [reply]
Yes I have used the following command to update to latest version
```
foundryup-zksync
zkstackup
```

**Reply by sarahschwartz** [reply]
Thanks for that. I'm curious if you have a `ZK_HOME` env variable  (or it's linked to some other env variable)? From a quick search it looks like another tool uses that, and I'm wondering if there is some interference. I'm not able to reproduce the issue on my end unfortunately.

**Reply by thomas-nguy** [reply]
I have tried to reinstall everything but no success.

Can you tell me which version I should be expected to use?

**Reply by thomas-nguy** [reply]
```
forge 0.0.2 (4a81422 2024-12-13T00:43:00.412610000Z)
cast 0.0.2 (4a81422 2024-12-13T00:43:00.483595000Z)
```

**Reply by sarahschwartz** [reply]
@thomas-nguy are you able to test in another environment in order to confirm if your environment is affecting this?

**Reply by kaymomin** [reply]
hi @thomas-nguy just checking if you were able to get it running?

**Reply by thomas-nguy** [reply]
unfortunately I did not had time to try it in another environment.

I am closing the issue as it seems I am the only one that is facing this issue (most likely env related)
